src/os/bluestore/BlueFS.cc

```diff
@@ -1477,6 +1477,10 @@ int BlueFS::_read(
     if (off < buf->bl_off || off >= buf->get_buf_end()) {
       s_lock.unlock();
       std::unique_lock u_lock(h->lock);
+      if (off >= buf->bl_off && off < buf->get_buf_end()) {
+        // if precondition has changed during locking upgrade.
```

Comment:
need s_lock.lock() before the continue here, right?

Reply:
no, we're under exclusive locking for the same h->lock mutex here. More than enough IMO..

Reply:
the continue will mean that u_lock goes out of scope and we drop the lock, though...

Reply:
yeah, my bad.. fixed.